pkg/providers/instance/instance.go

```diff
@@ -81,36 +81,33 @@ type Resource = map[string]interface{}
 
 type Provider struct {
 	location               string
-	azClient               *AZClient
+	AZClient               *AZClient
```

Comment:
Juan lee when I first started working on karpenter was really against the GetNic() pattern. Seems like if we need access to the clients, we should just expose the clients. This way the helpers stay decoupled and can be used in other projects or other places. 

You create the one client you need, and pass it into the armutil helper rather than having to fully configure a AZClient when using the code elsewhere.

Reply:
LMK if you have a different opinion or feel more strongly

Reply:
I don't have a strong opinion. Just mostly felt anti-pattern with the other clients since we have the others private, and now just `NetworkInterfacesClient` public. However, if we want to move that direction, and slowly transition it makes sense.

Reply:
Oh, never mind, for some reason the other public ones were hidden for me. This isn't an outlier. Mb